sync-core/src/main/java/com/cloudant/sync/replication/PullReplication.java

```diff
@@ -54,6 +60,66 @@ public PullReplication() {
         /* Does nothing but we can now document it */
     }
 
+    /**
+     * Adds request filters to run for every request made by this replication
+     * @param requestFilters The filters to run
+     * @return The current instance of PullReplication
+     */
+    public PullReplication addRequestFilters(List<HttpConnectionRequestFilter> requestFilters){
```

Comment:
No, Well we could, but the subclass would still need an implementation so you can seamless use the builder pattern. So we'd have something like:

```
public abstract Replication {

    public Replication addRequestFilters(HttpConnectionRequestFilter ... filters){
        this.filters.addAll(Arrays.asList(filters));
    }
}

public class PullReplication {

    @Override
     public PullReplication addRequestFilters(HttpConnectionRequestFilter ... filters){
         super.addRequestFilters(filters);
     }
}

```
or we could end up with some really complicated reflection so we could type infer whats going on using `T` it would look something like: 

```java
public abstract Replication<T extends Replication> {

    public T addRequestFilters(HttpConnectionRequestFilter ... filters){
        this.filters.addAll(Arrays.asList(filters));
    }
}

public class PullReplication extends Replication<PullReplication> {

}

```
(I haven't actually tried it, but my research suggests this would work)

I though implementing it completely twice was the lesser of three evils.